lib/dialects/fdbsql/index.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+// FDB SQL Layer index
+// This file was adapted from the PostgreSQL index
+// -------
+var _        = require('lodash');
+var inherits = require('inherits');
+
+var Client  = require('../../client');
+var Promise = require('../../promise');
+
+var pg;
+
+// Always initialize with the "QueryBuilder" and "QueryCompiler"
+// objects, which extend the base 'lib/query/builder' and
+// 'lib/query/compiler', respectively.
+function Client_FDB(config) {
+  Client.apply(this, arguments);
+  if (config.returning) this.defaultReturning = config.returning;
+  if (config.debug) this.isDebugging = true;
+  if (config.connection) {
+    // If no port has been defined,
+    if (!config.connection.port)
+      // set the port to the SQL Layer's default port, 15432.
+      config.connection.port = 15432;
+    this.initDriver();
+    this.initRunner();
+    this.connectionSettings = config.connection;
+    this.initPool();
+    this.pool = new this.Pool(config.pool);
+  }
+}
+inherits(Client_FDB, Client);
+
+Client_FDB.prototype.dialect = 'fdbsql';
+
+// Lazy load the pg dependency, since we might just be using
+// the client to generate SQL strings.
+Client_FDB.prototype.initDriver = function() {
+  pg = pg || (function() {
+    try {
+      return require('pg');
```

Comment:
It's correct as is. FoundationDB actually uses the pg network protocol for convenience, so I was able to use the same dependency as the PostgreSQL adapter.